src/devices/cpu/sh/sh.cpp

```diff
@@ -1547,68 +1508,51 @@ void sh_common_execution::SUBC(uint32_t m, uint32_t n)
 /*  SUBV    Rm,Rn */
 void sh_common_execution::SUBV(uint32_t m, uint32_t n)
 {
-	int32_t dest, src, ans;
-
-	if ((int32_t) m_sh2_state->r[n] >= 0)
-		dest = 0;
-	else
-		dest = 1;
-	if ((int32_t) m_sh2_state->r[m] >= 0)
-		src = 0;
-	else
-		src = 1;
+	int32_t dest = BIT(m_sh2_state->r[n], 31);
+	int32_t src = BIT(m_sh2_state->r[m], 31);
 	src += dest;
+
 	m_sh2_state->r[n] -= m_sh2_state->r[m];
-	if ((int32_t) m_sh2_state->r[n] >= 0)
-		ans = 0;
-	else
-		ans = 1;
+
+	int32_t ans = BIT(m_sh2_state->r[n], 31);
 	ans += dest;
-	if (src == 1)
-	{
-		if (ans == 1)
-			m_sh2_state->sr |= SH_T;
-		else
-			m_sh2_state->sr &= ~SH_T;
-	}
+
+	if (src == 1 && ans == 1)
+		m_sh2_state->sr |= SH_T;
 	else
 		m_sh2_state->sr &= ~SH_T;
```

Comment:
This changes behaviour.

Previous behaviour:

| src | ans | action |
| --- | --- | --- |
| != 1 | != 1 | no-op |
| != 1 | == 1 | no-op |
| == 1 | != 1 | `m_sh2_state->sr &= ~SH_T` |
| == 1 | == 1 | `m_sh2_state->sr \|= SH_T` |

New behaviour:

| src | ans | action |
| --- | --- | --- |
| != 1 | != 1 | `m_sh2_state->sr &= ~SH_T` |
| != 1 | == 1 | `m_sh2_state->sr &= ~SH_T` |
| == 1 | != 1 | `m_sh2_state->sr &= ~SH_T` |
| == 1 | == 1 | `m_sh2_state->sr \|= SH_T` |

That is, previously it would only do the and/or if `src == 1`, while now it takes the `else` path whenever `src` is not equal to 1.

Is this intentional?

Reply:
Hmm, does it? This is the previous code:

```
if (src == 1)
{
	if (ans == 1)
		m_sh2_state->sr |= SH_T;
	else
		m_sh2_state->sr &= ~SH_T;
}
else
	m_sh2_state->sr &= ~SH_T;
```

It was always doing `&= ~SH_T` if `src` was not equal to 1.

Reply:
Ah damn, I missed that the extra `else` branch was already there.